docs/core/integration-quality-scale/rules/iqs001.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "IQS001: Config flow"
+---
+
+IQS001 is as follows:
+
+Integration needs to be able to be set up via the UI
+
+## Reasoning
+Since its introduction in 2018, the config flow has become the standard way to set up integrations in Home Assistant.
+They allow for a consistent user experience across integrations and provide a way to guide users through the setup process.
+
+Because of the increased user experience, we want to make sure that all integrations are able to set up via the config flow.
+
+## Example implementation
+
+To use a config flow in your integration, you need to create a `config_flow.py` file in your integration folder and set `config_flow` in your `manifest.json` to `true`.
+
+`config_flow.py`:
+```python
+"""Config flow for my integration."""
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+
+class MyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """My config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input:
+            return self.async_create_entry(
+                title="MyIntegration",
+                data=user_input,
+            )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
+            errors=errors,
+        )
+```
+
+## Additional resources
+More information about config flows can be found in the [config flow documentation](../../../config_entries_index.md).
+
+## Exceptions
+There are no exceptions to this rule.
+
+## Related rules
+- IQS002 - Test a connection in the config flow
+- IQS003 - Don't allow the same device or service to be able to be set up twice
+- IQS004 - Full test coverage for the config flow
+- IQS037 - Keep the config flow understandable and validate where needed
+- IQS019 - Is discoverable when possible
+- IQS020 - Reauthentication
```

Comment:
```suggestion
- IQS020 - Reauthentication flow
```

Reply:
I'll update that one when I am at that rule. I also saw another confusion in the rules